clickhouse-admin/src/context.rs

```diff
@@ -33,18 +36,34 @@ impl ServerContext {
 
 pub struct SingleServerContext {
     clickhouse_cli: ClickhouseCli,
+    oximeter_client: OximeterClient,
     initialization_lock: Arc<Mutex<()>>,
 }
 
 impl SingleServerContext {
     pub fn new(clickhouse_cli: ClickhouseCli) -> Self {
-        Self { clickhouse_cli, initialization_lock: Arc::new(Mutex::new(())) }
+        let ip = clickhouse_cli.listen_address.ip();
+        let address = SocketAddrV6::new(*ip, CLICKHOUSE_TCP_PORT, 0, 0);
+        let log = clickhouse_cli
+            .log
+            .as_ref()
+            .expect("should have configured logging via CLI or config");
```

Comment:
This seems fishy to me; can we make `clickhouse_cli.log` not optional so we don't need this unwrap? Or alternatively, could we accept a `Logger` that we pass through to the client instead of relying on getting it from `clickhouse_cli`?

Reply:
It seemed fishy because it was; thanks. 1c282f93f makes `ClickhouseCli.log` non-optional.